service/csv/csv.go

```diff
@@ -0,0 +1,190 @@
+package csv
+
+import (
+	"encoding/csv"
+	"errors"
+	"log"
+	"os"
+	"strconv"
+	"sync"
+
+	"pokemons/model"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Service for Pokemon requests
+type Service struct {
+	logger *logrus.Logger
+	csvr   *os.File
+	csvw   *csv.Writer
+}
+
+// Task - Represents the task for every worker
+type Task struct {
+	ID   int
+	Data model.PokemonsData
+}
+
+// Worker - Worker struct
+type Worker struct {
+	task Task
+}
+
+var (
+	pokemons []*model.PokemonsData
+	rejected = 0
+	accepted = 0
+)
+
+// New creates a new PokemonAPI client
+func New(
+	logger *logrus.Logger,
+	csvr *os.File,
+	csvw *csv.Writer) *Service {
+	return &Service{logger, csvr, csvw}
+}
+
+// GetPokemonsInfo - Gets the pokemons info from the csv file
+func (s *Service) GetPokemonsInfo() ([]*model.PokemonsData, error) {
+	var pokemons []*model.PokemonsData
+
+	csvr := csv.NewReader(s.csvr)
+	data, err := csvr.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, row := range data {
+		pokemonID, err := strconv.Atoi(row[0])
+		if err != nil {
+			return nil, err
+		}
+		pokemonName := row[1]
+
+		pokemonWeight, err := strconv.Atoi(row[2])
+		if err != nil {
+			return nil, err
+		}
+
+		pokemons = append(pokemons, &model.PokemonsData{ID: pokemonID, Name: pokemonName,
+			Weight: pokemonWeight})
+	}
+	s.csvr.Seek(0, 0)
+
+	return pokemons, nil
+}
+
+// GetPokemonInfo -  Gets the pokemon info by ID from the csv file
+func (s *Service) GetPokemonInfo(pokemonID string) (*model.PokemonsData, error) {
+	var pokemons *model.PokemonsData
+
+	csvr := csv.NewReader(s.csvr)
+	data, err := csvr.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	for _, row := range data {
+		if pokemonID == row[0] {
+			rowID, err := strconv.Atoi(row[0])
+			if err != nil {
+				return nil, err
+			}
+
+			pokemonName := row[1]
+
+			pokemonWeight, err := strconv.Atoi(row[2])
+			if err != nil {
+				return nil, err
+			}
+
+			pokemons = &model.PokemonsData{ID: rowID, Name: pokemonName, Weight: pokemonWeight}
+
+			return pokemons, nil
+		}
+	}
+	s.csvr.Seek(0, 0)
+
+	return nil, errors.New("Pokemon not found")
+}
+
+// Work - Gets the pokemons concurrrently according to the filters
+func (s *Service) Work(validItems int, numberOfWorkers int, pokemonsInfo []*model.PokemonsData, typeOfItem string, perWorker int) <-chan string {
```

Comment:
Make sure to group common data types on your arguments, so that you can save some space `(validItems, numberOfWorkers, perWorker int, ...)`

Reply:
Done